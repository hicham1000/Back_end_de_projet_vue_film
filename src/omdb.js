import fetch from 'node-fetch';

/**
 * Cette fonction récupère les informations d'un film selon le titre spécifié
 * @param {string} title 
 * @returns {Promise<object|undefined>}
 */
export async function getMovieByTitle(title = '') {
}

/**
 * Cette fonction filtre :
 * - Les valeurs "undefined"
 * - Si la prop. "Metascore" est "undefined"
 * - Si la prop. "Metascore" est égale à "N/A"
 * @param {Array<object>} list 
 * @returns {Array<object>}
 */
export function getFilteredMovies(list = []) {
}

/**
 * Cette fonction retourne la liste des films ordonnés selon la clé spécifiée
 * @param {Array<object>} list 
 * @param {string} key 
 * @returns {Array<object>}
 */
export function getMoviesSortedBy(list = [], key = ''){
}

/**
 * Cette fonction retourne le top des films selon la clé spécifiée et le nombre demandé
 * @param {Array<object>} list 
 * @param {string} key 
 * @param {number} count 
 * @returns {Array<object>}
 */
export function getTopMoviesBy(list = [], key = '', count = 5){
}

/**
 * Cette fonction retourne un tableau de valeurs uniques d'une des prop. de la liste
 * @param {Array<object>} list 
 * @param {string} key 
 * @returns {Array<any>}
 */
export function getMovieListProp(list = [], key = ''){
}

/**
 * Cette fonction retourne le total d'une des proprietés de la liste
 * @param {Array<object>} list 
 * @param {string} key 
 * @returns {number}
 */
export function getMoviesSumOf(list = [], key = '') {
}

/**
 * Cette fonction retourne un tableau de tableau dont :
 * - le premier élément est le timestamp de passage du film et
 * - dont le deuxième élément est l'objet film
 * ex:
 * [
 *  [ 1234567890, { Title: '' } ],
 *  [ 1234567890, { Title: '' } ],
 *  [ 1234567890, { Title: '' } ],
 * ]
 * @param {Array<object>} list 
 * @param {number} from 
 * @returns {Array<object>}
 */
export function getReadyToPlayMoviesList(list = [], from = Date.now()){
}