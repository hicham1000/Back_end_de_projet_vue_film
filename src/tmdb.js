import fetch from 'node-fetch';

/**
 * Cette fonction retourne une liste de film de l'api themoviedb.org
 * @param {string|number} list 
 * @returns {Array<object>}
 */
export async function getMoviesList(list = '1') {
}

/**
 * Cette fonction retourne le nom des films d'une liste provenant de themoviedb.org
 * @param {Array<object>} list 
 * @returns {Array<string>}
 */
export function getMovieListTitles(list = []) {
}