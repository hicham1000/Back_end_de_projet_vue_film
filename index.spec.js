import fs from 'fs';
import util from 'util';

import * as tmdb from './src/tmdb.js';
import * as omdb from './src/omdb.js';

const toJson = (json) => JSON.stringify(json, null, 2)
const toFile = (...args) => util.promisify(fs.writeFile)(...args)
const fromFile = (...args) => util.promisify(fs.readFile)(...args)
const fromJson = (json) => JSON.parse(json)

// await toFile('./data.json', toJson(await tmdb.getMoviesList()));

// const titles = tmdb.getMovieListTitles(await tmdb.getMoviesList());
// const movies = omdb.getFilteredMovies(await Promise.all(titles.map(omdb.getMovieByTitle)));
// await toFile('./movies.omdb.json', toJson(movies));

describe('tmdb.js', () => {
    let movies;
    beforeEach(async () => {
        movies = fromJson(await fromFile('./movies.tmdb.json'));
    });
    
    describe("Le module doit exporter une function nommée", () => {
        test.each([
            'getMoviesList',
            'getMovieListTitles'
        ])("%s", fxname => {
            expect(fxname in tmdb).toBe(true);
            expect(tmdb[fxname] instanceof Function).toBe(true);
        })
    });

    test('getMoviesListTitle doit retourner tous les titres de la liste', async () => {
        const titles = tmdb.getMovieListTitles(movies);
        expect(titles).toEqual(expect.arrayContaining(['Spider-Man: No Way Home']));
    });
})

describe('omdb.js', () => {
    let list, movies;

    beforeEach(async () => {
        list = fromJson(await fromFile('./movies.tmdb.json'));
        movies = fromJson(await fromFile('./movies.omdb.json'));
    });

    describe("Le module doit exporter une function nommée :", () => {
        test.each([
            'getMovieByTitle',
            'getFilteredMovies',
            'getMoviesSortedBy',
            'getMovieListProp',
            'getTopMoviesBy',
            'getMoviesSumOf',
            'getReadyToPlayMoviesList'
        ])("%s", fxname => {
            expect(fxname in omdb).toBe(true);
            expect(omdb[fxname] instanceof Function).toBe(true);
        });
    })

    describe("'getFilteredMovies' doit retourner la liste des films n'ayant pas la prop. 'Metascore' égale à 'N/A' ou undefined", () => {
        test("getFilteredMovies(movies)", () => {
            const results = omdb.getFilteredMovies(movies);
            expect(results.map(m => m.Title)).toEqual(expect.not.arrayContaining(['X-Men: Apocalypse', 'X-Men: Days of Future Past']));
        })
    });

    describe("'getMovieByTitle' doit retourner le film", () => {
        test("Si la prop. 'Response' est égale à 'False', retourner undefined", async () => {
            const movie = await omdb.getMovieByTitle('ECN-Wiztivi-22');
            expect(movie).toBe(undefined);
        })
        test.each([
            'Spider-Man: No Way Home',
        ])("getMovieByTitle('%s')", async () => {
            const [ first ] = list;
            const movie = await omdb.getMovieByTitle(first.title);
            expect(movie).toMatchObject({
                Title: 'Spider-Man: No Way Home',
                Year: 2021,
                Rated: 'PG-13',
                Genre: ['Action', 'Adventure', 'Fantasy'],
                Director: ['Jon Watts'],
                Writer: ['Chris McKenna', 'Erik Sommers', 'Stan Lee'],
                Actors: ['Tom Holland', 'Zendaya', 'Benedict Cumberbatch'],
                Language: ['English'],
                Country: ['United States'],
                Metascore: 71,
                imdbRating: 8.5,
                imdbVotes: 555922,
                Type: 'movie',
                BoxOffice: 800588139,
            });
        })
    });

    describe('', () => {
        let movies;
        beforeEach(async () => {
            movies = omdb.getFilteredMovies(fromJson(await fromFile('./movies.omdb.json')));
        })
        describe("'getMoviesSortedBy' doit retourner les films ordonnés en fonction de la clé et du type de la valeur à traiter", () => {
            test.each([
                ['Title', 'string'],
                ['Year', 'number'],
                ['Rated', 'string'],
                ['Metascore', 'number'],
                ['imdbRating', 'number'],
            ])("getMoviesSortedBy(movies, '%s', '%s')",
            (key, type) => {
                const [ firstChar, ...restChar ] = key;
                expect(omdb.getMovieListProp(omdb.getMoviesSortedBy(movies, key, type), 'Title'))
                .toMatchSnapshot(`moviesSortedBy${firstChar.toUpperCase() + restChar.join()}`);
            });
        })
    
        describe("'getMovieListProp' doit retourner toutes les valeurs unique des films en fonction de la clé", () => {
            test.each([
                ['Genre', 'genres'],
                ['Director', 'directors'],
                ['Writer', 'writer'],
                ['Actors', 'actors'],
                ['Language', 'language'],
                ['Country', 'country'],
                ['Type', 'types'],
            ])("getMovieListProp(movies, '%s')", (key, name) => {
                expect(omdb.getMovieListProp(movies, key)).toMatchSnapshot(name);
            });
        })
    
        describe("'getTopMoviesBy' doit retourner le top des films du plus grand au plus petit en fonction de la clé et du type de la valeur à traiter", () => {
            test.each([
                ['Metascore'],
                ['imdbRating'],
            ])("getTopMoviesBy(movies, '%s')", (key) => {
                expect(omdb.getMovieListProp(omdb.getTopMoviesBy(movies, key, 5), 'Title')).toMatchSnapshot();
            })
        })
    
        describe("'getMoviesSumOf' doit retourner la valeur totale d'une propriété des films en fonction de la clé et du type de valeur à traiter", () => {
            test.each([
                ['imdbRating', 'string'],
                ['BoxOffice', 'string'],
            ])("getMoviesSumOf(movies, '%s', '%s')", (key, type) => {
                expect(omdb.getMoviesSumOf(movies, key, type)).toMatchSnapshot(key + 'Sum');
            })
        })
    })
})