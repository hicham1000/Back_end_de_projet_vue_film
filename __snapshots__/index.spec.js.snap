// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omdb.js  'getMovieListProp' doit retourner toutes les valeurs unique des films en fonction de la clé getMovieListProp(movies, 'Actors'): actors 1`] = `
Array [
  "Tom Holland",
  "Zendaya",
  "Benedict Cumberbatch",
  "Gemma Chan",
  "Richard Madden",
  "Angelina Jolie",
  "Simu Liu",
  "Awkwafina",
  "Tony Chiu-Wai Leung",
  "Scarlett Johansson",
  "Florence Pugh",
  "David Harbour",
  "Samuel L. Jackson",
  "Jake Gyllenhaal",
  "Robert Downey Jr.",
  "Chris Evans",
  "Mark Ruffalo",
  "Brie Larson",
  "Ben Mendelsohn",
  "Paul Rudd",
  "Evangeline Lilly",
  "Michael Peña",
  "Chris Hemsworth",
  "Chadwick Boseman",
  "Michael B. Jordan",
  "Lupita Nyong'o",
  "Tom Hiddleston",
  "Cate Blanchett",
  "Michael Keaton",
  "Chris Pratt",
  "Zoe Saldana",
  "Dave Bautista",
  "Hugh Jackman",
  "Patrick Stewart",
  "Dafne Keen",
  "Chiwetel Ejiofor",
  "Rachel McAdams",
  "Ryan Reynolds",
  "Morena Baccarin",
  "T.J. Miller",
  "Ioan Gruffudd",
  "Michael Chiklis",
  "Michael Douglas",
  "Corey Stoll",
  "Vin Diesel",
  "Bradley Cooper",
  "Andrew Garfield",
  "Emma Stone",
  "Jamie Foxx",
  "Natalie Portman",
  "Will Yun Lee",
  "Tao Okamoto",
  "Guy Pearce",
  "Gwyneth Paltrow",
  "Rhys Ifans",
  "Nicolas Cage",
  "Ciarán Hinds",
  "Idris Elba",
  "Hugo Weaving",
  "James McAvoy",
  "Michael Fassbender",
  "Jennifer Lawrence",
  "Anthony Hopkins",
  "Mickey Rourke",
  "Liev Schreiber",
  "Ray Stevenson",
  "Dominic West",
  "Julie Benz",
  "Edward Norton",
  "Liv Tyler",
  "Tim Roth",
  "Terrence Howard",
  "Tobey Maguire",
  "Kirsten Dunst",
  "Topher Grace",
  "Eva Mendes",
  "Sam Elliott",
  "Halle Berry",
  "Jennifer Garner",
  "Goran Visnjic",
  "Wesley Snipes",
  "Kris Kristofferson",
  "Parker Posey",
  "Alfred Molina",
  "Eric Bana",
  "Jennifer Connelly",
  "Willem Dafoe",
  "Ron Perlman",
  "Ian McKellen",
  "Stephen Dorff",
  "Lea Thompson",
  "Jeffrey Jones",
  "Tim Robbins",
]
`;

exports[`omdb.js  'getMovieListProp' doit retourner toutes les valeurs unique des films en fonction de la clé getMovieListProp(movies, 'Country'): country 1`] = `
Array [
  "United States",
  "Australia",
  "Canada",
  "Germany",
  "United Kingdom",
  "Japan",
  "United Arab Emirates",
  "Switzerland",
]
`;

exports[`omdb.js  'getMovieListProp' doit retourner toutes les valeurs unique des films en fonction de la clé getMovieListProp(movies, 'Director'): directors 1`] = `
Array [
  "Jon Watts",
  "Chloé Zhao",
  "Destin Daniel Cretton",
  "Cate Shortland",
  "Anthony Russo",
  "Joe Russo",
  "Anna Boden",
  "Ryan Fleck",
  "Peyton Reed",
  "Ryan Coogler",
  "Taika Waititi",
  "James Gunn",
  "James Mangold",
  "Scott Derrickson",
  "Tim Miller",
  "Tim Story",
  "Joss Whedon",
  "Marc Webb",
  "Alan Taylor",
  "Shane Black",
  "Mark Neveldine",
  "Brian Taylor",
  "Joe Johnston",
  "Matthew Vaughn",
  "Kenneth Branagh",
  "Jon Favreau",
  "Gavin Hood",
  "Lexi Alexander",
  "Louis Leterrier",
  "Sam Raimi",
  "Mark Steven Johnson",
  "Brett Ratner",
  "Rob Bowman",
  "David S. Goyer",
  "Ang Lee",
  "Bryan Singer",
  "Guillermo del Toro",
  "Stephen Norrington",
  "Willard Huyck",
]
`;

exports[`omdb.js  'getMovieListProp' doit retourner toutes les valeurs unique des films en fonction de la clé getMovieListProp(movies, 'Genre'): genres 1`] = `
Array [
  "Action",
  "Adventure",
  "Fantasy",
  "Sci-Fi",
  "Drama",
  "Comedy",
  "Thriller",
  "Crime",
  "Horror",
]
`;

exports[`omdb.js  'getMovieListProp' doit retourner toutes les valeurs unique des films en fonction de la clé getMovieListProp(movies, 'Language'): language 1`] = `
Array [
  "English",
  "American Sign ",
  "Marathi",
  "Spanish",
  "Sumerian",
  "Latin",
  "Greek",
  "Ancient (to 1453)",
  "Mandarin",
  "Russian",
  "Norwegian",
  "Hungarian",
  "Macedonian",
  "Finnish",
  "Italian",
  "Czech",
  "Japanese",
  "Xhosa",
  "German",
  "French",
  "Swahili",
  "Nama",
  "Korean",
  "Romanian",
  "Hindi",
  "Portuguese",
  "Persian",
  "Urdu",
  "Arabic",
  "Kurdish",
  "Esperanto",
  "Chinese",
  "Serbian",
]
`;

exports[`omdb.js  'getMovieListProp' doit retourner toutes les valeurs unique des films en fonction de la clé getMovieListProp(movies, 'Type'): types 1`] = `
Array [
  "movie",
]
`;

exports[`omdb.js  'getMovieListProp' doit retourner toutes les valeurs unique des films en fonction de la clé getMovieListProp(movies, 'Writer'): writer 1`] = `
Array [
  "Chris McKenna",
  "Erik Sommers",
  "Stan Lee",
  "Chloé Zhao",
  "Patrick Burleigh",
  "Ryan Firpo",
  "Dave Callaham",
  "Destin Daniel Cretton",
  "Andrew Lanham",
  "Eric Pearson",
  "Jac Schaeffer",
  "Ned Benson",
  "Christopher Markus",
  "Stephen McFeely",
  "Anna Boden",
  "Ryan Fleck",
  "Geneva Robertson-Dworet",
  "Paul Rudd",
  "Ryan Coogler",
  "Joe Robert Cole",
  "Craig Kyle",
  "Christopher L. Yost",
  "Jonathan Goldstein",
  "John Francis Daley",
  "Jon Watts",
  "James Gunn",
  "Dan Abnett",
  "Andy Lanning",
  "James Mangold",
  "Scott Frank",
  "Michael Green",
  "Jon Spaihts",
  "Scott Derrickson",
  "C. Robert Cargill",
  "Joe Simon",
  "Rhett Reese",
  "Paul Wernick",
  "Mark Frost",
  "Michael France",
  "Edgar Wright",
  "Joe Cornish",
  "Adam McKay",
  "Joss Whedon",
  "Jack Kirby",
  "Nicole Perlman",
  "Alex Kurtzman",
  "Roberto Orci",
  "Jeff Pinkner",
  "Mark Bomback",
  "Drew Pearce",
  "Shane Black",
  "James Vanderbilt",
  "Alvin Sargent",
  "Steve Kloves",
  "Zak Penn",
  "Scott M. Gimple",
  "Seth Hoffman",
  "David S. Goyer",
  "Ashley Miller",
  "Zack Stentz",
  "Jane Goldman",
  "Don Payne",
  "Justin Theroux",
  "Don Heck",
  "David Benioff",
  "Skip Woods",
  "Nick Santora",
  "Art Marcum",
  "Matt Holloway",
  "Mark Fergus",
  "Hawk Ostby",
  "Sam Raimi",
  "Ivan Raimi",
  "Mark Steven Johnson",
  "Simon Kinberg",
  "Frank Miller",
  "Marv Wolfman",
  "Gene Colan",
  "Steve Ditko",
  "Alfred Gough",
  "James Schamus",
  "David Hayter",
  "Bryan Singer",
  "David Koepp",
  "Tom DeSanto",
  "Steve Gerber",
  "Willard Huyck",
  "Gloria Katz",
]
`;

exports[`omdb.js  'getMoviesSortedBy' doit retourner les films ordonnés en fonction de la clé et du type de la valeur à traiter getMoviesSortedBy(movies, 'Metascore', 'number'): moviesSortedByMe,t,a,s,c,o,r,e 1`] = `
Array [
  "Howard the Duck",
  "Punisher: War Zone",
  "Ghost Rider: Spirit of Vengeance",
  "Elektra",
  "Ghost Rider",
  "Blade: Trinity",
  "Fantastic Four",
  "X-Men Origins: Wolverine",
  "Blade",
  "Eternals",
  "Blade II",
  "The Amazing Spider-Man 2",
  "Thor: The Dark World",
  "Hulk",
  "Thor",
  "Iron Man 2",
  "X-Men: The Last Stand",
  "Spider-Man 3",
  "The Wolverine",
  "The Incredible Hulk",
  "Iron Man 3",
  "Captain Marvel",
  "Ant-Man",
  "X-Men",
  "Deadpool",
  "X-Men: First Class",
  "Avengers: Age of Ultron",
  "The Amazing Spider-Man",
  "Captain America: The First Avenger",
  "Black Widow",
  "Guardians of the Galaxy Vol. 2",
  "Avengers: Infinity War",
  "X2: X-Men United",
  "Spider-Man: Far from Home",
  "The Avengers",
  "Ant-Man and the Wasp",
  "Captain America: The Winter Soldier",
  "Spider-Man: No Way Home",
  "Shang-Chi and the Legend of the Ten Rings",
  "Doctor Strange",
  "Spider-Man: Homecoming",
  "Spider-Man",
  "Thor: Ragnarok",
  "Captain America: Civil War",
  "Guardians of the Galaxy",
  "Logan",
  "Avengers: Endgame",
  "Iron Man",
  "Spider-Man 2",
  "Black Panther",
]
`;

exports[`omdb.js  'getMoviesSortedBy' doit retourner les films ordonnés en fonction de la clé et du type de la valeur à traiter getMoviesSortedBy(movies, 'Rated', 'string'): moviesSortedByRa,t,e,d 1`] = `
Array [
  "Howard the Duck",
  "Spider-Man: No Way Home",
  "Eternals",
  "Shang-Chi and the Legend of the Ten Rings",
  "Black Widow",
  "Spider-Man: Far from Home",
  "Avengers: Endgame",
  "Captain Marvel",
  "Ant-Man and the Wasp",
  "Avengers: Infinity War",
  "Black Panther",
  "Thor: Ragnarok",
  "Spider-Man: Homecoming",
  "Guardians of the Galaxy Vol. 2",
  "Doctor Strange",
  "Captain America: Civil War",
  "Fantastic Four",
  "Ant-Man",
  "Avengers: Age of Ultron",
  "Guardians of the Galaxy",
  "The Amazing Spider-Man 2",
  "Captain America: The Winter Soldier",
  "Thor: The Dark World",
  "The Wolverine",
  "Iron Man 3",
  "The Amazing Spider-Man",
  "The Avengers",
  "Ghost Rider: Spirit of Vengeance",
  "Captain America: The First Avenger",
  "X-Men: First Class",
  "Thor",
  "Iron Man 2",
  "X-Men Origins: Wolverine",
  "The Incredible Hulk",
  "Iron Man",
  "Spider-Man 3",
  "Ghost Rider",
  "X-Men: The Last Stand",
  "Elektra",
  "Spider-Man 2",
  "Hulk",
  "X2: X-Men United",
  "Spider-Man",
  "X-Men",
  "Logan",
  "Deadpool",
  "Punisher: War Zone",
  "Blade: Trinity",
  "Blade II",
  "Blade",
]
`;

exports[`omdb.js  'getMoviesSortedBy' doit retourner les films ordonnés en fonction de la clé et du type de la valeur à traiter getMoviesSortedBy(movies, 'Title', 'string'): moviesSortedByTi,t,l,e 1`] = `
Array [
  "Ant-Man",
  "Ant-Man and the Wasp",
  "Avengers: Age of Ultron",
  "Avengers: Endgame",
  "Avengers: Infinity War",
  "Black Panther",
  "Black Widow",
  "Blade",
  "Blade II",
  "Blade: Trinity",
  "Captain America: Civil War",
  "Captain America: The First Avenger",
  "Captain America: The Winter Soldier",
  "Captain Marvel",
  "Deadpool",
  "Doctor Strange",
  "Elektra",
  "Eternals",
  "Fantastic Four",
  "Ghost Rider",
  "Ghost Rider: Spirit of Vengeance",
  "Guardians of the Galaxy",
  "Guardians of the Galaxy Vol. 2",
  "Howard the Duck",
  "Hulk",
  "Iron Man",
  "Iron Man 2",
  "Iron Man 3",
  "Logan",
  "Punisher: War Zone",
  "Shang-Chi and the Legend of the Ten Rings",
  "Spider-Man",
  "Spider-Man 2",
  "Spider-Man 3",
  "Spider-Man: Far from Home",
  "Spider-Man: Homecoming",
  "Spider-Man: No Way Home",
  "The Amazing Spider-Man",
  "The Amazing Spider-Man 2",
  "The Avengers",
  "The Incredible Hulk",
  "The Wolverine",
  "Thor",
  "Thor: Ragnarok",
  "Thor: The Dark World",
  "X-Men",
  "X-Men Origins: Wolverine",
  "X-Men: First Class",
  "X-Men: The Last Stand",
  "X2: X-Men United",
]
`;

exports[`omdb.js  'getMoviesSortedBy' doit retourner les films ordonnés en fonction de la clé et du type de la valeur à traiter getMoviesSortedBy(movies, 'Year', 'number'): moviesSortedByYe,a,r 1`] = `
Array [
  "Howard the Duck",
  "Blade",
  "X-Men",
  "Spider-Man",
  "Blade II",
  "Hulk",
  "X2: X-Men United",
  "Blade: Trinity",
  "Spider-Man 2",
  "Fantastic Four",
  "Elektra",
  "X-Men: The Last Stand",
  "Spider-Man 3",
  "Ghost Rider",
  "Punisher: War Zone",
  "The Incredible Hulk",
  "Iron Man",
  "X-Men Origins: Wolverine",
  "Iron Man 2",
  "Ghost Rider: Spirit of Vengeance",
  "Captain America: The First Avenger",
  "X-Men: First Class",
  "Thor",
  "The Amazing Spider-Man",
  "The Avengers",
  "Thor: The Dark World",
  "The Wolverine",
  "Iron Man 3",
  "Guardians of the Galaxy",
  "The Amazing Spider-Man 2",
  "Captain America: The Winter Soldier",
  "Ant-Man",
  "Avengers: Age of Ultron",
  "Doctor Strange",
  "Captain America: Civil War",
  "Deadpool",
  "Thor: Ragnarok",
  "Spider-Man: Homecoming",
  "Guardians of the Galaxy Vol. 2",
  "Logan",
  "Ant-Man and the Wasp",
  "Avengers: Infinity War",
  "Black Panther",
  "Spider-Man: Far from Home",
  "Avengers: Endgame",
  "Captain Marvel",
  "Spider-Man: No Way Home",
  "Eternals",
  "Shang-Chi and the Legend of the Ten Rings",
  "Black Widow",
]
`;

exports[`omdb.js  'getMoviesSortedBy' doit retourner les films ordonnés en fonction de la clé et du type de la valeur à traiter getMoviesSortedBy(movies, 'imdbRating', 'number'): moviesSortedByIm,d,b,R,a,t,i,n,g 1`] = `
Array [
  "Ghost Rider: Spirit of Vengeance",
  "Elektra",
  "Howard the Duck",
  "Ghost Rider",
  "Hulk",
  "Fantastic Four",
  "Blade: Trinity",
  "Punisher: War Zone",
  "Spider-Man 3",
  "Eternals",
  "The Amazing Spider-Man 2",
  "X-Men Origins: Wolverine",
  "Black Widow",
  "The Wolverine",
  "The Incredible Hulk",
  "X-Men: The Last Stand",
  "Blade II",
  "Captain Marvel",
  "Thor: The Dark World",
  "The Amazing Spider-Man",
  "Captain America: The First Avenger",
  "Ant-Man and the Wasp",
  "Thor",
  "Iron Man 2",
  "Blade",
  "Iron Man 3",
  "Black Panther",
  "Ant-Man",
  "Avengers: Age of Ultron",
  "Shang-Chi and the Legend of the Ten Rings",
  "Spider-Man: Far from Home",
  "Spider-Man: Homecoming",
  "Spider-Man 2",
  "X2: X-Men United",
  "Spider-Man",
  "X-Men",
  "Doctor Strange",
  "Guardians of the Galaxy Vol. 2",
  "X-Men: First Class",
  "Captain America: Civil War",
  "Captain America: The Winter Soldier",
  "Thor: Ragnarok",
  "Iron Man",
  "Deadpool",
  "Logan",
  "Guardians of the Galaxy",
  "The Avengers",
  "Avengers: Endgame",
  "Avengers: Infinity War",
  "Spider-Man: No Way Home",
]
`;

exports[`omdb.js  'getMoviesSumOf' doit retourner la valeur totale d'une propriété des films en fonction de la clé et du type de valeur à traiter getMoviesSumOf(movies, 'BoxOffice', 'string'): BoxOfficeSum 1`] = `14020046534`;

exports[`omdb.js  'getMoviesSumOf' doit retourner la valeur totale d'une propriété des films en fonction de la clé et du type de valeur à traiter getMoviesSumOf(movies, 'imdbRating', 'string'): imdbRatingSum 1`] = `355.1999999999999`;

exports[`omdb.js  'getTopMoviesBy' doit retourner le top des films du plus grand au plus petit en fonction de la clé et du type de la valeur à traiter getTopMoviesBy(movies, 'Metascore') 1`] = `
Array [
  "Black Panther",
  "Spider-Man 2",
  "Iron Man",
  "Avengers: Endgame",
  "Logan",
]
`;

exports[`omdb.js  'getTopMoviesBy' doit retourner le top des films du plus grand au plus petit en fonction de la clé et du type de la valeur à traiter getTopMoviesBy(movies, 'imdbRating') 1`] = `
Array [
  "Spider-Man: No Way Home",
  "Avengers: Infinity War",
  "Avengers: Endgame",
  "The Avengers",
  "Guardians of the Galaxy",
]
`;
